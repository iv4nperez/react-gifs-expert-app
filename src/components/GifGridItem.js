import React from 'react'

export const GifGridItem = ({ id, title, url }) => {

    return (
        <div className="card animate__animated animate__fadeIn">
           <img className="cover" src={ url } alt={ title } />
           <p>{ title }</p>
        </div>
    )
}
