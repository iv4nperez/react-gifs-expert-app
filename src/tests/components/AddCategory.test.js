import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

import { AddCategory } from '../../components/AddCategory';

describe('Pruebas en <AddCategory />', () => {
    const setCategories = jest.fn();
    let wraper = shallow( <AddCategory setCategories={ setCategories }  /> );

    beforeEach(()=>{
        jest.clearAllMocks();
        wraper = shallow( <AddCategory setCategories={ setCategories }  /> );
    });

    test('debe de mostrase correctamente', () => {
        expect( wraper ).toMatchSnapshot();
    });

    test('debe de cambiar la caja de texto', () => {
       
        const input = wraper.find('input');
        const value = 'Hola mundo';
        input.simulate('change', {  target: { value } });
        
    });
    
    test('no debe de postear la informacion onSubmit', () => {
       
        wraper.find('form').simulate('submit', { preventDefault(){} });
        expect( setCategories ).not.toHaveBeenCalled();
    });
    
    test('debe de llamar el setCategories y limpiar la caja de texto', () => {
       
        const input = wraper.find('input');
        const value = 'Hola mundo';
        input.simulate('change', {  target: { value } });

        wraper.find('form').simulate('submit', { preventDefault(){} });
        
        expect( setCategories ).toHaveBeenCalled();

        expect( input.prop('value').trim() ).toBe( '' );
    });
    

});