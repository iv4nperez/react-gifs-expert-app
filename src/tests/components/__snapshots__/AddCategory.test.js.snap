// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> debe de mostrase correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Busca tu GIF favorito"
    type="text"
    value=""
  />
</form>
`;
